### PR TITLE
add-prom-metrics

### PR DESCRIPTION
**Pull request by kaisbaccour**
This is a dashboard for the deployment of this PR.  https://shorturl.at/vxJN4
Screenshot 2024-03-01 at 20 15 26

Once merged I will set the dashboard to point to the main branch

### CODE REVIEW

**Comment by CyberHoward** [comment]
Lol, completely forgot we can just create multiple runtimes!!

File: bot/src/bot.rs [hunk 1]
```diff
@@ -59,6 +60,11 @@ pub struct Bot {
     // Autocompound information
     contract_instances_to_ac: HashSet<(String, Addr)>,
     pub autocompound_cooldown: Duration,
+    // metrics
+    fetch_count: IntCounter,
+    fetch_instances_count: IntGauge,
+    autocompounded_count: IntCounter,
+    autocompounded_error_count: IntCounter,
```
Comment (Buckram123) [inline bot/src/bot.rs:67]: Would be cleaner to get it into separate struct

File: bot/src/bot.rs [hunk 2]
```diff
@@ -100,12 +142,15 @@ impl Bot {
...
         )?;
+        let mut fetch_instances_count = 0;
 
         for app_info in saving_modules.modules {
             let code_id = app_info.module.reference.unwrap_app()?;
+
             let mut contract_addrs = daemon
                 .rt_handle
                 .block_on(utils::fetch_instances(daemon.channel(), code_id))?;
+            fetch_instances_count += contract_addrs.len();
```
Comment (Buckram123) [inline bot/src/bot.rs:153]: Count added before retain not sure if we want to include contracts without permission! You can set it to the length of `contract_instances_to_autocompound` after the loop if we don't want to include those

**Review by Buckram123 — APPROVED** [review]
Very nice!!

**Review by Kayanski — APPROVED** [review]

File: bot/src/bot.rs (hunk 1 above)
Comment (kaisbaccour) [reply bot/src/bot.rs]: Yeah good point! 
It's on a different struct now.

File: bot/src/bot.rs (hunk 2 above)
Comment (kaisbaccour) [reply bot/src/bot.rs:166]: This is for now just to get the tooling for metrics. Then we need to make a session where we ask what we want to track and adapt accordingly.

**Pull request merged**